src/main/java/org/opentripplanner/index/IndexGraphQLSchema.java

```diff
@@ -2033,26 +2033,44 @@ private void createPlanType(GraphIndex index) {
                 .name("date")
                 .description("The time and date of travel")
                 .type(Scalars.GraphQLLong)
-                .dataFetcher(environment -> ((TripPlan)environment.getSource()).date.getTime())
+                .dataFetcher(environment -> ((TripPlan) ((Map)environment.getSource()).get("plan")).date.getTime())
                 .build())
             .field(GraphQLFieldDefinition.newFieldDefinition()
                 .name("from")
                 .description("The origin")
                 .type(new GraphQLNonNull(placeType))
-                .dataFetcher(environment -> ((TripPlan)environment.getSource()).from)
+                .dataFetcher(environment -> ((TripPlan) ((Map)environment.getSource()).get("plan")).from)
                 .build())
             .field(GraphQLFieldDefinition.newFieldDefinition()
                 .name("to")
                 .description("The destination")
                 .type(new GraphQLNonNull(placeType))
-                .dataFetcher(environment -> ((TripPlan)environment.getSource()).to)
+                .dataFetcher(environment -> ((TripPlan) ((Map)environment.getSource()).get("plan")).to)
                 .build())
             .field(GraphQLFieldDefinition.newFieldDefinition()
                 .name("itineraries")
                 .description("A list of possible itineraries")
                 .type(new GraphQLNonNull(new GraphQLList(itineraryType)))
-                .dataFetcher(environment -> ((TripPlan)environment.getSource()).itinerary)
+                .dataFetcher(environment -> ((TripPlan) ((Map)environment.getSource()).get("plan")).itinerary)
                 .build())
-             .build();
+            .field(GraphQLFieldDefinition.newFieldDefinition()
+                .name("messages")
+                .description("A list of possible error messages")
+                .type(new GraphQLNonNull(new GraphQLList(Scalars.GraphQLString)))
```

Comment:
Isn't this a bit too simple to have just a String here? In the UI we probably want to translate the error messages, or modify them somehow (e.g. map certain messages to the same text). Therefore, it would be nice to have some other error code. The type of the Exception could work as a code.